### PR TITLE
[ GRU ] Skeleton Code for GRU

### PR DESCRIPTION
**Pull request by jijoongmoon**
This commit includes,
 . Skeleton Code for GRU Layer

**Self evaluation:**
1. Build test:	 [X]Passed [ ]Failed [ ]Skipped
2. Run test:	 [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: jijoong.moon <jijoong.moon@samsung.com>

### CODE REVIEW

**Review by zhoonit — APPROVED** [review]
LGTM. How about cherry pick this PR on top of main? I think we can merge faster.

File: nntrainer/layers/gru.cpp [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+
+#define NUM_GATE 3
+
+// - weight_xh ( input to hidden )
+//  : [1, 1, input_size, unit (hidden_size) x NUM_GATE] -> f, g, i, o
+// - weight_hh ( hidden to hidden )
+//  : [1, 1, unit (hidden_size) , unit (hidden_size) x NUM_GATE] -> f, g, i, o
+// - bias_h ( hidden bias )
+//  : [1, 1, 1, unit (hidden_size) x NUM_GATE] -> f, g, i, o
+int GRULayer::initialize(Manager &manager) {
```
Comment (kparichay) [inline nntrainer/layers/gru.cpp:37]: can you please consider reutilizing LSTM/RNN code for GRU, and reducing code for all 3. 
code maintenance will be tough if all 3 stay independent.

Comment (jijoongmoon) [reply nntrainer/layers/gru.cpp:37]: let me think. thanks.

**Review by kparichay — APPROVED** [review]
LGTM

**Review by zhoonit — APPROVED** [review]
Thank you 😺

**Pull request merged**